.github/workflows/CI.yml

```diff
@@ -325,49 +341,53 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
 
+      - uses: pnpm/action-setup@v3
+        with:
+          version: 9.0.2
+
       - name: Cache dependencies
         id: cache
         uses: actions/cache@v3
         with:
           path: '**/node_modules'
-          key: ${{ matrix.node }}-${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+          key: ${{ matrix.node }}-${{ runner.os }}-modules-${{ hashFiles('**/pnpm-lock.yaml') }}
 
       - name: Add node-gyp
-        run: yarn global add node-gyp
+        run: npm install -g node-gyp
 
       - name: Install dependencies
-        run: yarn --ignore-engines
+        run: pnpm install
 
-      - name: Assert yarn.lock is correct
+      - name: Assert pnpm.lock is correct
         run: |
           if ! git diff --quiet; then
-            echo 'You need to update the yarn.lock file (potentially after running: yarn global add node-gyp)'
+            echo 'pnpm lock file is incorrect'
             exit 1
           fi
 
       - name: Build
-        run: yarn build
+        run: pnpm run build
 
       - name: Run linter
         env:
           GITHUB_BASE_REF: ${{ github.base_ref }}
-        run: yarn lint-branch
+        run: pnpm run lint-branch
 
       # To disable eslint-diff just comment the following two lines
       #- name: Run eslint-diff
       #  run: yarn workspace commonwealth run lint-diff
 
       - name: Run eslint-diff-canary to make sure eslint-diff is functioning properly
-        run: yarn workspace commonwealth run lint-diff-canary
+        run: cd packages/commonwealth && pnpm run lint-diff-canary
 
       - name: Commonwealth Typecheck
-        run: yarn --cwd packages/commonwealth check-types
+        run: cd packages/commonwealth && pnpm run check-types
```

Comment:
`check-types` is a standard script that all packages must implement - just one pnpm --filter should run them all, even in parallel

Reply:
I'm just trying to keep what we had before.

I'll discuss this with you guys but I think this PR should *just* switch us to pnpm. Nothing else.  Otherwise it's going to be a massive PR and take forever. 

I still have some things to fix.  The PR isn't ready yet.